### PR TITLE
Rail Recipe App

### PR DESCRIPTION
**Pull request by the-faizmohammad**
**Pull Request Summary**

### Pages

1. **Login Page**
   - Built with Devise for authentication.
   - Personalized design for enhanced user experience.

2. **Registration Page**
   - Utilizes Devise for user registration.
   - Custom design elements for a cohesive look.

3. **Recipes List**
   - Displays recipes created by the logged-in user.
   - Provides links to individual recipe details.
   - Implemented with attention to avoiding N+1 queries.

4. **Public Recipe List**
   - Presents a catalog of public recipes, sorted by newest.
   - Users can delete recipes they own.
   - Linked to recipe details for exploration.

5. **Recipe Details**
   - Features a toggle button for setting recipe visibility.
   - Shows recipe details for public recipes or those owned by the user.
   - Owners can access a form for adding new food items.

### Navigation Menu
- Offers convenient access to all pages created.
- Ensures smooth navigation for users.

### Tests
- Includes comprehensive unit and integration tests for robust functionality.

### Exclusive Features (For groups with less than 3 members)

6. **Food List**
   - Lists food items added by the logged-in user, including quantity.
   - Allows seamless addition of new food items.

7. **General Shopping List View**
   - Compares all recipes with the user's food list to generate a shopping list.
   - Displays total items needed and their cumulative price.

---

This pull request encompasses the development of essential pages, features, and tests for the project, ensuring a user-friendly experience and robust functionality.

Screenshot 2024-02-16 200153

### CODE REVIEW

File: spec/features/food/food_new_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    @f1 = Food.create(user: @user, name: 'Food item', measurement_unit: 'kg', price: 2, quantity: 5)
+  end
+
+  context 'Adds new Food item' do
+    it 'Instance to the table' do
+      visit new_user_session_path
+      fill_in 'Email', with: @user.email
+      fill_in 'Password', with: @user.password
+      click_button 'Log in'
+      sleep(1)
```
Comment (omar-muhamad) [inline spec/features/food/food_new_spec.rb:21]: - [ ] Some test cases are failing, Please ensure that all test cases are passing.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @the-faizmohammad and @Izebiafe,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/food/food_new_spec.rb (hunk 1 above)
Comment (the-faizmohammad) [reply spec/features/food/food_new_spec.rb:21]: Screenshot 2024-02-16 200153

**Review by OyePriscilla — APPROVED** [review]
Hi @the-faizmohammad and @Izebiafe 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
All written tests passed ✔️ 

Untitled

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**